adapters/yeahmobi/yeahmobi.go

```diff
@@ -0,0 +1,179 @@
+package yeahmobi
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/golang/glog"
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/macros"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"net/http"
+	"text/template"
+)
+
+type YeahmobiAdapter struct {
+	EndpointTemplate template.Template
+}
+
+func NewYeahmobiBidder(endpointTemplate string) adapters.Bidder {
+	tpl, err := template.New("endpointTemplate").Parse(endpointTemplate)
+	if err != nil {
+		glog.Fatal("Unknow url template")
+		return nil
+	}
+	return &YeahmobiAdapter{EndpointTemplate: *tpl}
+}
+
+func (adapter *YeahmobiAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var adapterRequests []*adapters.RequestData
+
+	adapterRequest, errors := adapter.makeRequest(request)
+	if adapterRequest != nil {
+		adapterRequests = append(adapterRequests, adapterRequest)
+	}
+
+	return adapterRequests, errors
+}
+
+func (adapter *YeahmobiAdapter) makeRequest(request *openrtb.BidRequest) (*adapters.RequestData, []error) {
+	var errs []error
+
+	yeahmobiExt, errs := getYeahmobiExt(request)
+
+	if yeahmobiExt == nil {
+		glog.Fatal("Invalid ExtImpYeahmobi value")
+		return nil, errs
+	}
+
+	endPoint, err := adapter.getEndpoint(yeahmobiExt)
+
+	transform(request)
+
+	if err != nil {
+		return nil, append(errs, err)
+	}
+	reqBody, err := json.Marshal(request)
+
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+
+	return &adapters.RequestData{
+		Method:  "POST",
+		Uri:     endPoint,
+		Body:    reqBody,
+		Headers: headers,
+	}, errs
+}
+
+func transform(request *openrtb.BidRequest) {
+	for i, imp := range request.Imp {
+		if imp.Native != nil {
+			var nativeRequest map[string]interface{}
+			err := json.Unmarshal([]byte(request.Imp[i].Native.Request), &nativeRequest)
+			if err == nil {
+				if nativeRequest["native"] != nil {
```

Comment:
You would typically use the second return parameter. It's a boolean which indicates if the key was found in the map.